smart_open/smart_open_lib.py

```diff
@@ -28,11 +28,23 @@
 import requests
 import io
 
+from functools import partial
+
 
 IS_PY2 = (sys.version_info[0] == 2)
 
 if IS_PY2:
     import httplib
+    if sys.version_info[1] == 6:
+        import copy_reg
```

Comment:
@piskvorky 
How about this: https://github.com/shaform/smart_open/commit/3154b060ef6e6388b76848ccb23ac029d0b183a4.